pkg/operator/controller/controller_default_certificate.go

```diff
@@ -62,11 +66,18 @@ func (r *reconciler) ensureDefaultCertificateForIngress(caSecret *corev1.Secret,
 // desiredRouterDefaultCertificateSecret returns the desired default certificate
 // secret.
 func desiredRouterDefaultCertificateSecret(ca *crypto.CA, deployment *appsv1.Deployment, ci *ingressv1alpha1.ClusterIngress) (*corev1.Secret, error) {
+	// Without an ingress domain, we cannot generate a default certificate.
+	if len(ci.Status.IngressDomain) == 0 {
```

Comment:
If in the calling reconciliation function we're enforcing `status.ingressDomain` persistence before we even process the clusteringress further, should this instead return an error indicating we violated our invariant (i.e. a bug)?

Reply:
We could get `len(ci.Status.IngressDomain) == 0` from an explicit empty value, so this is not necessarily an error.